src/showingpreviously/cinemas/jackroe.py

```diff
@@ -0,0 +1,163 @@
+
+WTWCINEMAS_BASE_URL = 'https://wtwcinemas.co.uk'
+WTWCINEMAS_CHAIN = Chain('WTW Cinemas')
+WTWCINEMAS_LOCATION_CODE_PATTERN = re.compile(r'/websales/sales/(?P<location_code>.+?)/book')
+
+
+def get_response(url: str) -> requests.Response:
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise CinemaArchiverException(f'Got status code {r.status_code} when fetching URL {url}')
+    return r
+
+
+def get_film_attributes(film_title: str) -> dict[str, any]:
+    attributes = {}
+    if 'SUBTITLED SCREENING' in film_title:
+        attributes['subtitled'] = True
+        film_title = film_title.replace('SUBTITLED SCREENING', '')
+    if '(DUBBED)' in film_title:
+        attributes['language'] = 'English'
+        film_title = film_title.replace('(DUBBED)', '')
+    if film_title.endswith('2D'):
+        attributes['format'] = [' 2D']
+        film_title = film_title[:-3]
+    if film_title.endswith('3D'):
+        attributes['format'] = [' 3D']
+        film_title = film_title[:-3]
+    return film_title, attributes
+
+
+def get_attributes(html: str) -> dict[str, any]:
+    attributes = {}
+    if 'Audio Description' in html:
+        attributes['audio-described'] = True
+    if 'Subtitled' in html:
+        attributes['subtitled'] = True
+    return attributes
+
+
+def get_showings_date(cinema_id: str, cinema: Cinema, chain: Chain, showing_dates: [str]) -> [Showing]:
+    url = SHOWING_LIST_URL.format(cinema_code=cinema_id)
+    r = get_response(url)
+    soup = BeautifulSoup(r.text, features='html.parser')
+    showings = []
+    for movie_row in soup.find_all('div', {'class': 'start-performance-box'}):
+        film_title = movie_row.find('h3').text.strip()
+        film_title = film_title.replace('\n', ' ').replace('\t', ' ')
+        film_title = re.sub(' +', ' ', film_title)
+        film_title, film_attributes = get_film_attributes(film_title)
+        film = Film(film_title, UNKNOWN_FILM_YEAR)
+        for day_row in movie_row.find_all('div', {'class': 'col-md-4'}):
+            day = day_row.text.strip()
+            if day not in showing_dates:
+                continue
+            booking_btn_row = day_row.find_next('div', {'class': 'col-md-8'})
+            for booking_btn in booking_btn_row.find_all('a', {'class': 'btn'}):
+                booking_link = booking_btn['href']
+                showing_code = SHOWING_CODE_PATTERN.search(booking_link).group('showing_code')
+                url = SHOWING_URL.format(cinema_code=cinema_id, showing_code=showing_code)
+                r = get_response(url)
+                if FINISHED_TEXT in r.text:
+                    # the screening has already happened, and we can't get the screen any more
+                    # skip it
+                    continue
+                showing_info = SHOWING_INFO_PATTERN.search(r.text)
+                screen = Screen(showing_info.group('screen_name').replace('  ', ' '))
```

Comment:
Be aware that one of the cinemas has a ascreen called `Screen 2 NEW`. This is something that we should edit manually in the DB afterwards, though.